src/rag/vector_store/pgvector.py

```
@@ -1,28 +1,383 @@
+                updated_at = CURRENT_TIMESTAMP
+        """).format(table=sql.Identifier(self.table_name))
+
+        with self._get_connection() as conn:
+            with conn.cursor() as cur:
+                for i in range(0, len(processed_docs), batch_size):
+                    batch_docs = processed_docs[i:i + batch_size]
+                    batch_embeddings = embeddings[i:i + batch_size]
 
-    def add_documents(self, documents: List[Dict[str, Any]]) -> None:
-        raise NotImplementedError("pgvector add_documents is not implemented yet")
+                    for doc, embedding in zip(batch_docs, batch_embeddings):
```

Ruff (0.15.0)

[warning] `zip()` without an explicit `strict=` parameter

Add explicit value for parameter `strict=`

(B905)